### PR TITLE
New testcase: External Aero Model, Generic F1 Rear Wing

### PR DESCRIPTION
**Pull request by agvarghe**

### CODE REVIEW

File: examples/gallery/lucid_generic_f1_rear_wing.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+# Connect the PyPrimeMesh client and get the model.
+# Instantiate meshing utilities from the ``lucid`` class.
+
+import os
+import tempfile
+
+import ansys.meshing.prime as prime
+from ansys.meshing.prime.graphics import Graphics
+
+prime_client = prime.launch_prime(timeout=40)
```
Comment (waltersma) [inline examples/gallery/lucid_generic_f1_rear_wing.py:49]: @agvarghe we need to discuss why this is needed as we need to fix the underlying issues

File: examples/gallery/lucid_generic_f1_rear_wing.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+Procedure
+~~~~~~~~~~
+* Launch an Ansys Prime Server instance and instantiate the meshing utilities
+  from the ``lucid`` class.
+* Import the geometries.
+* Merge parts.
+* Perform connect operation between the compoents.
+* Surface mesh with curvature sizing.
+* Extract Fluid Region by defining a material pount.
```
Comment (waltersma) [inline examples/gallery/lucid_generic_f1_rear_wing.py:28]:
```suggestion
* Extract fluid region by defining a material point.
```

File: examples/gallery/lucid_generic_f1_rear_wing.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+from ansys.meshing.prime.graphics import Graphics
+
+prime_client = prime.launch_prime(timeout=40)
+model = prime_client.model
+mesh_util = prime.lucid.Mesh(model=model)
+
+###############################################################################
+# Import geometry
+# ~~~~~~~~~~~~~~~
+# Download the Rear Wing geometries (stl) file exported by SpaceClaim.
```
Comment (waltersma) [inline examples/gallery/lucid_generic_f1_rear_wing.py:56]:
```suggestion
# Download the rear wing geometries (STL's).
```

File: examples/gallery/lucid_generic_f1_rear_wing.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+f1_rw_drs = prime.examples.download_f1_rw_drs_stl()
+
+f1_rw_enclosure = prime.examples.download_f1_rw_enclosure_stl()
+
+f1_rw_end_plates = prime.examples.download_f1_rw_end_plates_stl()
+
+f1_rw_main_plane = prime.examples.download_f1_rw_main_plane_stl()
+
+wkspFldr = os.path.split(os.path.abspath(f1_rw_main_plane))[0]
```
Comment (waltersma) [inline examples/gallery/lucid_generic_f1_rear_wing.py:69]: @agvarghe I am not sure this is a given that all files will be in the same folder or this does not seem like a clean way to do this to me.  Any thoughts @ninad-kamat ?

File: examples/gallery/lucid_generic_f1_rear_wing.py (hunk 1 above)
Comment (agvarghe) [reply examples/gallery/lucid_generic_f1_rear_wing.py:49]: This i will remove , when running locally i added the timeout definition

File: examples/gallery/lucid_generic_f1_rear_wing.py (hunk 4 above)
Comment (agvarghe) [reply examples/gallery/lucid_generic_f1_rear_wing.py:69]: Sure, this was just an experiment, on my local machine, it is downloaded in the same folder

File: src/ansys/meshing/prime/examples/examples.py [hunk 5]
```diff
@@ -782,3 +791,139 @@ def download_pcb_pmdat(
...
+    str
+        Local path to the downloaded file
+
+    Examples
+    --------
+    >>> import ansys.meshing.prime as prime
+    >>> import ansys.meshing.prime.examples as prime_examples
+    >>> with prime.launch_prime() as session:
+    >>>     model = session.model
+    >>>     pcb = prime_examples.download_pcb_pmdat()
```
Comment (waltersma) [inline src/ansys/meshing/prime/examples/examples.py:821]: @agvarghe need to fix examples.

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+
+prime_client = prime.launch_prime()
+model = prime_client.model
+mesh_util = prime.lucid.Mesh(model=model)
+
+###############################################################################
+# Import geometry
+# ~~~~~~~~~~~~~~~
+# Download the Generic F1 rear wing geometries (stl) files exported from SpaceClaim.
+# Import each geometry from the example folder.
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:63]:
```suggestion
# Import each geometry and append to the model.
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+f1_rw_end_plates = prime.examples.download_f1_rw_end_plates_stl()
+
+f1_rw_main_plane = prime.examples.download_f1_rw_main_plane_stl()
+
+for file_name in [f1_rw_drs, f1_rw_enclosure, f1_rw_end_plates, f1_rw_main_plane]:
+    mesh_util.read(file_name, append=True)
+
+display = Graphics(model)
+scope = prime.ScopeDefinition(model, part_expression="* !*enclosure*")
+
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:80]: needs display

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+# ~~~~~~~~~~~~~~~~~~~
+# In order to accurately represent the physics of the DRS wing, a limitation of 8 mm
+# is imposed on the mesh size of the wing.
+# This is accomplished by implementing curvature size control, which refines the
+# mesh according to the curvature of the DRS surfaces.
+# Additionally, to accurately capture the curved surfaces of other sections of the
+# wing, curvature control is defined with a normal angle of 18 degrees.
+# A volumetric size field is then computed based on the defined size control.
+
+# Local Curvature size control on all components
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:133]: Not that local ;)  Local control everywhere...

Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:133]: Please review these controls and comments, what is used and where, and the comments on these controls explaining them.  Avoid use of terms such as component as this seems as though you are referring to parts.

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+    prime.ScopeDefinition(
+        model, evaluation_type=prime.ScopeEvaluationType.ZONES, zone_expression="* !f1_rw_enclosure"
+    )
+)
+
+
+###############################################################################
+# Define Prism Controls
+# ~~~~~~~~~~~~~~~~~~~
+# Prism control can be used to define inflation layers on the external aero surfaces.
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:216]:
```suggestion
# A prism control can be used to define inflation layers on the external aero surfaces.
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+# Specify the aero surfaces using labels, here prism scope is defined on zones associated
+# with labels ``*drs*`` and ``*plane*``.
+# The growth for the prism layer is control by defining offset type to
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:219]:
```suggestion
# The growth for the prism layer is controlled by defining the offset type to
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+        n_layers=5,
+        first_height=0.5,
+        growth_rate=1.2,
+    )
+)
+
+###############################################################################
+# Generate Volume Mesh
+# ~~~~~~~~~~~~~~~~~~~
+# Volume mesh with polyhedral elements and boundary layer refinement.
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:252]:
```suggestion
# Volume mesh with hexcore polyhedral elements and boundary layer refinement.
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+**Summary**: This example showcases the process of generating a mesh for a generic F1 rear wing
+STL file model.
+
+Objective
+~~~~~~~~~~
+
+The given example demonstrates the interconnection between various elements of the rear wing of
+a generic F1 car, along with volume meshes, utilizing a poly-hexcore mesh containing prisms.To
+simplify the process and enhance convenience, multiple meshing utilities provided in the
+"lucid" class are employed.
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:17]:
```suggestion
The example demonstrates how to connect various parts of a rear wing of
a generic F1 car and volume mesh the resulting model using a poly-hexcore mesh containing prisms.
To simplify the process and enhance convenience, multiple meshing utilities provided in the
"lucid" class are used.
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+.. image:: ../../../images/generic_rear_wing.png
+   :align: center
+   :width: 800
+   :alt: Generic F1 rear wing.
+
+Procedure
+~~~~~~~~~~
+* Launch an Ansys Prime Server instance and instantiate the meshing utilities
+  from the ``lucid`` class.
+* Import the STL geometry file for each component of the F1 rear wing.
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:28]:
```suggestion
* Import and append the STL geometry files for each part of the F1 rear wing.
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 14]
```diff
@@ -0,0 +1,314 @@
+"""
+.. _ref_generic_f1_rw:
+
+================================================================
+Meshing a Generic F1 car rear wing for external aero simulation
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:5]:
```suggestion
Meshing a generic F1 car rear wing for external aero simulation
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+from ansys.meshing.prime.graphics import Graphics
+
+prime_client = prime.launch_prime()
+model = prime_client.model
+mesh_util = prime.lucid.Mesh(model=model)
+
+###############################################################################
+# Import geometry
+# ~~~~~~~~~~~~~~~
+# Download the Generic F1 rear wing geometries (stl files).
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:62]:
```suggestion
# Download the generic F1 rear wing geometries (stl files).
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 16]
```diff
@@ -0,0 +1,314 @@
...
+f1_rw_drs = prime.examples.download_f1_rw_drs_stl()
+f1_rw_enclosure = prime.examples.download_f1_rw_enclosure_stl()
+f1_rw_end_plates = prime.examples.download_f1_rw_end_plates_stl()
+f1_rw_main_plane = prime.examples.download_f1_rw_main_plane_stl()
+
+for file_name in [f1_rw_drs, f1_rw_enclosure, f1_rw_end_plates, f1_rw_main_plane]:
+    mesh_util.read(file_name, append=True)
+
+display = Graphics(model)
+
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:75]:
```suggestion

# display the rear wing geometry without the enclosure
display = Graphics(model)
scope = prime.ScopeDefinition(model, part_expression="* !*enclosure*")
display(scope=scope)

```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 17]
```diff
@@ -0,0 +1,314 @@
...
+###############################################################################
+# Merge Parts
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:77]:
```suggestion
# Merge parts
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 18]
```diff
@@ -0,0 +1,314 @@
...
+# ~~~~~~~~~~~~~~~~~~~
+# Establish the Global Size parameter to regulate mesh refinement.
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:79]:
```suggestion
# Establish the global size parameter to regulate mesh refinement.
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+# Merge all individual parts into a unified part named `f1_car_rear_wing`.
+
+# Define Global sizes
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:82]:
```suggestion
# Define global sizes
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 20]
```diff
@@ -0,0 +1,314 @@
...
+model.set_global_sizing_params(prime.GlobalSizingParams(model, min=4, max=32, growth_rate=1.2))
+
+# Create label per part
+for part in model.parts:
+    part.add_labels_on_zonelets([part.name.split(".")[0]], part.get_face_zonelets())
+
+# Merge Parts
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:89]:
```suggestion
# Merge parts
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 21]
```diff
@@ -0,0 +1,314 @@
...
+    prime.SurfaceDiagnosticSummaryParams(
+        model,
+        compute_free_edges=True,
+        compute_self_intersections=True,
+    )
+)
+print(f"Total number of free edges present is {surf_report.n_free_edges}")
+
+###############################################################################
+# Define local size-control and Generate size-field
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:118]:
```suggestion
# Define local size-control and generate size-field
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 22]
```diff
@@ -0,0 +1,314 @@
...
+# ~~~~~~~~~~~~~~~~~~~
+# In order to accurately represent the physics of the DRS wing, a limitation of 8 mm
+# is imposed on the mesh size of the wing.
+# This is accomplished by implementing curvature size control, which refines the
+# mesh according to the curvature of the DRS surfaces.
+# Additionally, to accurately capture the curved surfaces of other sections of the
+# wing, curvature control is defined with a normal angle of 18 degrees.
+# A volumetric size field is then computed based on the defined size controls.
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:126]: how is this used?  surface meshing and/or volume meshing?

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 23]
```diff
@@ -0,0 +1,314 @@
...
+
+# local Curvature size control for DRS
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:128]:
```suggestion
# Local curvature size control for DRS
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 24]
```diff
@@ -0,0 +1,314 @@
...
+curv_scope = prime.ScopeDefinition(
+    model,
+    entity_type=prime.ScopeEntity.FACEZONELETS,
+    part_expression="f1_car_rear_wing*",
+    label_expression="*drs*",
+)
+curv_size_control.set_scope(curv_scope)
+curv_size_control.set_suggested_name("curvature_drs")
+
+# Global Curvature size control on all face zones of the rear wing
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:141]:
```suggestion
# Global curvature size control on all face zones of the rear wing
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 25]
```diff
@@ -0,0 +1,314 @@
...
+# Compute volumetric sizefield
+compute_size = prime.SizeField(model)
+vol_sf_params = prime.VolumetricSizeFieldComputeParams(model)
+compute_size.compute_volumetric(
+    [curv_size_control.id, curv_size_control_global.id], volumetric_sizefield_params=vol_sf_params
+)
+
+
+###############################################################################
+# Generate Surface Mesh
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:162]:
```suggestion
# Generate surface mesh
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 26]
```diff
@@ -0,0 +1,314 @@
...
+for part in model.parts:
+    part.add_labels_on_zonelets([part.name.split(".")[0]], part.get_face_zonelets())
+
+# Merge parts
+merge_params = prime.MergePartsParams(model, merged_part_suggested_name="f1_car_rear_wing")
+merge_result = model.merge_parts([part.id for part in model.parts], merge_params)
+part = model.get_part_by_name(merge_result.merged_part_assigned_name)
+
+###############################################################################
+# Mesh Connect
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:95]:
```suggestion
# Mesh connect
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 27]
```diff
@@ -0,0 +1,314 @@
...
+
+###############################################################################
+# Compute Volumetric Regions
+# ~~~~~~~~~~~~~~~~~~~
+# Compute the volume zones.
+
+mesh_util.compute_volumes(part_expression=part.name, create_zones_per_volume=True)
+
+###############################################################################
+# Define Volume Controls
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:187]:
```suggestion
# Define volume controls
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 28]
```diff
@@ -0,0 +1,314 @@
...
+        model,
+        offset_type=prime.PrismControlOffsetType.UNIFORM,
+        n_layers=5,
+        first_height=0.5,
+        growth_rate=1.2,
+    )
+)
+
+###############################################################################
+# Generate Volume Mesh
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:247]:
```suggestion
# Generate volume mesh
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 29]
```diff
@@ -0,0 +1,314 @@
...
+results = search.get_volume_quality_summary(params=params)
+
+# Print statistics on meshed part
+print(part_summary_res)
+print(
+    "\nMaximum inverse-orthoginal quality of the Volume Mesh : ",
+    results.quality_results_part[0].max_quality,
+)
+
+# Mesh Check
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:288]:
```suggestion
# Mesh check
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 30]
```diff
@@ -0,0 +1,314 @@
...
+
+scope = prime.ScopeDefinition(model, part_expression="*", label_expression="* !*enclosure*")
+display(
+    scope=scope,
+)
+
+###############################################################################
+# Write mesh
+# ~~~~~~~~~~
+# Export as cas file for External Aero Simulations
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:300]:
```suggestion
# Export as cas file for external aero simulations
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 31]
```diff
@@ -0,0 +1,314 @@
...
+)
+
+# Mesh Check
+result = prime.VolumeMeshTool(model).check_mesh(part.id, params=prime.CheckMeshParams(model))
+print("\nMesh Check", result, sep="\n")
+
+scope = prime.ScopeDefinition(model, part_expression="*", label_expression="* !*enclosure*")
+display(
+    scope=scope,
+)
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:295]:
```suggestion
display(scope=scope)
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 32]
```diff
@@ -0,0 +1,314 @@
...
+#
+# Create a surface mesh for the rear wing using the defined size controls.
+# To facilitate the definition of boundary conditions on the surfaces in the solver,
+# generate face zones by utilizing the existing labels found in the rear wing model.
+
+mesh_util.surface_mesh_with_size_controls(size_control_names="*curvature*")
+scope = prime.ScopeDefinition(model, label_expression="* !*enclosure*")
+display(
+    scope=scope,
+)
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:173]:
```suggestion
display(scope=scope)
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 33]
```diff
@@ -0,0 +1,317 @@
...
+
+f1_rw_drs = prime.examples.download_f1_rw_drs_stl()
+f1_rw_enclosure = prime.examples.download_f1_rw_enclosure_stl()
+f1_rw_end_plates = prime.examples.download_f1_rw_end_plates_stl()
+f1_rw_main_plane = prime.examples.download_f1_rw_main_plane_stl()
+
+for file_name in [f1_rw_drs, f1_rw_enclosure, f1_rw_end_plates, f1_rw_main_plane]:
+    mesh_util.read(file_name, append=True)
+
+
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:74]:
```suggestion
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 34]
```diff
@@ -0,0 +1,316 @@
...
+curv_size_control_global.set_suggested_name("curvature_global")
+
+# Compute volumetric sizefield
+compute_size = prime.SizeField(model)
+vol_sf_params = prime.VolumetricSizeFieldComputeParams(model)
+compute_size.compute_volumetric(
+    [curv_size_control.id, curv_size_control_global.id], volumetric_sizefield_params=vol_sf_params
+)
+
+
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:166]:
```suggestion
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 35]
```diff
@@ -0,0 +1,315 @@
...
+compute_size = prime.SizeField(model)
+vol_sf_params = prime.VolumetricSizeFieldComputeParams(model)
+compute_size.compute_volumetric(
+    [curv_size_control.id, curv_size_control_global.id], volumetric_sizefield_params=vol_sf_params
+)
+
+###############################################################################
+# Generate surface mesh
+# ~~~~~~~~~~~~~~~~~~~
+#
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:169]:
```suggestion
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 36]
```diff
@@ -0,0 +1,314 @@
...
+# ~~~~~~~~~~~~~~~~~~~
+# Create a surface mesh for the rear wing using the defined size controls.
+# To facilitate the definition of boundary conditions on the surfaces in the solver,
+# generate face zones by utilizing the existing labels found in the rear wing model.
+
+mesh_util.surface_mesh_with_size_controls(size_control_names="*curvature*")
+scope = prime.ScopeDefinition(model, label_expression="* !*enclosure*")
+display(scope=scope)
+
+# Create face-zones per label
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:177]:
```suggestion
# Create face zones per label
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 37]
```diff
@@ -0,0 +1,314 @@
...
+mesh_util.surface_mesh_with_size_controls(size_control_names="*curvature*")
+scope = prime.ScopeDefinition(model, label_expression="* !*enclosure*")
+display(scope=scope)
+
+# Create face zones per label
+for label in part.get_labels():
+    mesh_util.create_zones_from_labels(label_expression=label)
+
+###############################################################################
+# Compute Volumetric Regions
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:182]:
```suggestion
# Compute volumetric regions
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 38]
```diff
@@ -0,0 +1,314 @@
...
+# Define volume controls
+# ~~~~~~~~~~~~~~~~~~~
+# In order to prevent the generation of a volume mesh within the solid wing,
+# the type of a volume zone within the rear wing can be defined as "dead".
+# To accomplish this, Volume Control is utilized to assign the type for the
+# specific volume zone.
+# Expressions are employed to define the volume zones that need to be filled, with
+# "* !f1_rw_enclosure" indicating that it applies to all volume zones except
+# for "f1_rw_enclosure".
+volume_control_1 = model.control_data.create_volume_control()
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:198]:
```suggestion

volume_control_1 = model.control_data.create_volume_control()
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 39]
```diff
@@ -0,0 +1,315 @@
...
+# ~~~~~~~~~~~~~~~~~~~
+# In order to prevent the generation of a volume mesh within the solid wing,
+# the type of a volume zone within the rear wing can be defined as "dead".
+# To accomplish this, Volume Control is utilized to assign the type for the
+# specific volume zone.
+# Expressions are employed to define the volume zones that need to be filled, with
+# "* !f1_rw_enclosure" indicating that it applies to all volume zones except
+# for "f1_rw_enclosure".
+
+volume_control_1 = model.control_data.create_volume_control()
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:199]:
```suggestion
volume_control = model.control_data.create_volume_control()
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 40]
```diff
@@ -0,0 +1,315 @@
...
+# In order to prevent the generation of a volume mesh within the solid wing,
+# the type of a volume zone within the rear wing can be defined as "dead".
+# To accomplish this, Volume Control is utilized to assign the type for the
+# specific volume zone.
+# Expressions are employed to define the volume zones that need to be filled, with
+# "* !f1_rw_enclosure" indicating that it applies to all volume zones except
+# for "f1_rw_enclosure".
+
+volume_control = model.control_data.create_volume_control()
+volume_control_1.set_params(
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:200]:
```suggestion
volume_control.set_params(
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 41]
```diff
@@ -0,0 +1,315 @@
...
+# for "f1_rw_enclosure".
+
+volume_control = model.control_data.create_volume_control()
+volume_control.set_params(
+    prime.VolumeControlParams(
+        model,
+        cell_zonelet_type=prime.CellZoneletType.DEAD,
+    )
+)
+volume_control_1.set_scope(
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:206]:
```suggestion
volume_control.set_scope(
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 42]
```diff
@@ -0,0 +1,315 @@
...
+        cell_zonelet_type=prime.CellZoneletType.DEAD,
+    )
+)
+volume_control.set_scope(
+    prime.ScopeDefinition(
+        model, evaluation_type=prime.ScopeEvaluationType.ZONES, zone_expression="* !f1_rw_enclosure"
+    )
+)
+
+
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:212]:
```suggestion
```

File: examples/gallery/08_lucid_generic_f1_rear_wing.py [hunk 43]
```diff
@@ -0,0 +1,314 @@
...
+# ~~~~~~~~~~~~~~~~~~~
+# Volume mesh with hexcore polyhedral elements and boundary layer refinement.
+
+volume_mesh = prime.AutoMesh(model)
+auto_mesh_param = prime.AutoMeshParams(
+    model,
+    prism_control_ids=[prism_control.id],
+    size_field_type=prime.SizeFieldType.VOLUMETRIC,
+    volume_fill_type=prime.VolumeFillType.HEXCOREPOLY,
+    volume_control_ids=[volume_control_1.id],
```
Comment (waltersma) [inline examples/gallery/08_lucid_generic_f1_rear_wing.py:259]:
```suggestion
    volume_control_ids=[volume_control.id],
```

**Review by waltersma — APPROVED** [review]

**Review by ninad-kamat — APPROVED** [review]

**Pull request merged**